### PR TITLE
make fused_multi_transformer support dynamically set the cache_kvs' shape and support input prefix_caches.

### PR DESCRIPTION
**Pull request by carryyu**
### PR types

Function optimization
### PR changes

OPs
### Describe

make fused_multi_transformer support dynamically set the cache_kvs' shape and support input prefix_caches.

### CODE REVIEW

File: paddle/fluid/operators/fused/fused_multi_transformer_int8_op.cu [hunk 1]
```diff
@@ -13,7 +13,7 @@ See the License for the specific language governing permissions and
 limitations under the License. */
 
 #include "paddle/fluid/operators/fused/attn_gemm_int8.h"
-#include "paddle/fluid/operators/fused/fused_multi_transformer_op.h"
+#include "paddle/fluid/operators/fused/fused_multi_transformer.h"
```
Comment (LielinJiang) [inline paddle/fluid/operators/fused/fused_multi_transformer_int8_op.cu:16]: why renamed this file?

Comment (carryyu) [reply paddle/fluid/operators/fused/fused_multi_transformer_int8_op.cu:16]: 为了通过pre-commit的格式检查，如果不rename在格式检查时会要求必须在fused_multi_transformer_op.cc中包含fused_multi_transformer_op.h文件，但实际是在fused_multi_transformer_op.cu文件中包含该头文件。
fused_multi_transformer_op.h由飞圣创建，与他讨论后修改该头文件名称。

**Review by Ligoml — DISMISSED** [review]
LGTM for docs
后续挪到正式目录下的时候注意增加一下中文API文档

**Comment by carryyu** [comment]
> 嗯嗯好的

**Pull request closed without merging**